File: 1.Easy/530. Minimum Absolute Difference in BST.py
class TreeNode:
    def __init__(self,val = 0, left = None, right = None) -> None:
        self.val = val 
        self.left = left 
        self.right = right
    
class Solution: 
    def getMinimumDifference(self, root : TreeNode) -> int :
        def push_node_in_array(root,array:list[int]):
            if not root: 
                return array
            array.append(root.val)
            array = push_node_in_array(root.left,array)
            array = push_node_in_array(root.right,array)

            return array
        arr = push_node_in_array(root,[])
        arr.sort()
        ans = 100001
        for i in range(len(arr)-1):
            ans = min(ans,arr[i+1] - arr[i])
        return ans
              

File: 1.Easy/530. Minimum Absolute Difference in BST.cpp
#include<iostream>
#include<vector>
using namespace std;

struct TreeNode{
	int val;
	TreeNode *left;
	TreeNode *right;
	TreeNode(): val(0), left(nullptr), right(nullptr){
	}
	TreeNode(int val):val(val),left(nullptr),right(nullptr){
	}
	TreeNode(int val, TreeNode*left, TreeNode*right): val(val), left(left), right(right){
	}
};

class Solution {
public:
    int min_dif = INT_MAX;
    int val = -1;
int getMinimumDifference(TreeNode* root) {
    if (root->left != NULL)
        getMinimumDifference(root->left);
    if (val >= 0)
        min_dif = min(min_dif, root->val - val);
    val = root->val;
    if (root->right != NULL)
        getMinimumDifference(root->right);
    return min_dif;
    }
};
int main(){

}
